test/ProblemDetailsResponseFactoryTest.php

```diff
@@ -176,4 +178,14 @@ public function testFactoryRendersPreviousExceptionsInDebugMode() : void
         $this->assertEquals(101010, $payload['exception']['stack'][0]['code']);
         $this->assertEquals('first', $payload['exception']['stack'][0]['message']);
     }
+
+    public function testFragileDataInExceptionMessageShouldBeHiddenInResponseBodyInNoneDebugMode()
+    {
+        $fragileMessage = 'Your SQL or password here';
+        $exception = new \Exception($fragileMessage);
+
+        $response = $this->factory->createResponseFromThrowable($this->request->reveal(), $exception);
+
+        $this->assertNotContains($fragileMessage, (string) $response->getBody());
```

Comment:
This is a tough call.

Generally, we want to include some usable detail here. If we _never_ include the exception message, then what _do_ we include in order to provide useful details to the end-user consuming the API?

I would argue that if you have exceptions that have sensitive details in them, you should be catching them, and doing one of the following:

- Using the `ProblemDetailsResponseFactory` to generate a custom response, via the `createResponse()` method.
- Creating a custom `ProblemDetailsException` type and re-throwing that type with the details you're interested in.

Alternately, we could have a flag for the `ProblemDetailsResponseFactory` to disable using the exception message for the `detail`; if we do that, we likely should have a _default detail message_ to use, however.